cwltool/main.py

```diff
@@ -112,7 +113,16 @@ def _terminate_processes() -> None:
     # It's possible that another thread will spawn a new task while
     # we're executing, so it's not safe to use a for loop here.
     while processes_to_kill:
-        processes_to_kill.popleft().kill()
+        process = processes_to_kill.popleft()
+        cidfile = [str(arg).split("=")[1] for arg in process.args if "--cidfile" in str(arg)]
+        if cidfile:
+            with open(cidfile[0], "r") as inp_stream:
+                p = subprocess.Popen(["docker", "kill", inp_stream.read()], shell=False)
+                try:
+                    p.wait(timeout=10)
+                except subprocess.TimeoutExpired:
+                    p.kill()
```

Comment:
```suggestion
            try:
                with open(cidfile[0], "r") as inp_stream:
                    p = subprocess.Popen(["docker", "kill", inp_stream.read()], shell=False)
                    try:
                        p.wait(timeout=10)
                    except subprocess.TimeoutExpired:
                        p.kill()
            except FileNotFoundError:
                pass
```
This came up in the conformance tests: https://github.com/common-workflow-language/cwltool/runs/2066391703?check_suite_focus=true#step:5:939